app/model/BaseRule.scala

```diff
@@ -0,0 +1,24 @@
+package model
+
+import play.api.libs.json.{JsObject, Json}
+
+/**
+  * A rule to match text against.
+  */
+trait BaseRule {
+  val id: String
+  val category: Category
+  val description: String
+  val suggestions: List[Suggestion]
+  val autoApplyFirstSuggestion: Boolean
+}
+
+object BaseRule {
+  def toJson(rule: BaseRule): JsObject = Json.obj(
```

Comment:
Could this be a `Writes[BaseRule]`? I think it would save writing out the formatter manually in `RuleMatch` below.

Reply:
BaseRule is a trait, so I get `No apply function found for model.BaseRule`, but if this were possible it'd be ideal -- I'll ask about.

Reply:
Ah yeah, I think you'd have to write all the fields out manually which wouldn't be much fun. No problem, it's fine as it is